### PR TITLE
[15.0][MIG] mig date_range

### PR DESCRIPTION
**Pull request by JoanMForgeFlow**
Supersede https://github.com/OCA/server-ux/pull/401

### CODE REVIEW

**Comment by HaraldPanten** [comment]
@pedrobaeza could you approve workflows here? THX.

**Comment by HaraldPanten** [comment]
@JoanMForgeFlow could you check pre-commit?

**Comment by JoanMForgeFlow** [comment]
> @JoanMForgeFlow could you check pre-commit?

done!

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍 

@ValentinVinagre @pedrobaeza

File: date_range/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Date Range",
     "summary": "Manage all kind of date range",
-    "version": "14.0.2.0.2",
+    "version": "15.0.1.0.1",
```
Comment (ValentinVinagre) [inline date_range/__manifest__.py]: Change to 15.0.1.0.0

Comment (JoanMForgeFlow) [reply date_range/__manifest__.py]: done!

**Review by ValentinVinagre — APPROVED** [review]

**Comment by HaraldPanten** [comment]
@pedrobaeza do you mind merging this one in order to continue with the migration of --> https://github.com/OCA/account-financial-reporting/pull/837 ?

THX!

**Review by AaronHForgeFlow — COMMENTED** [review]
Seems to work ok. Can you include the improvements of https://github.com/OCA/server-ux/pull/413 ?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Missing changes from #413

**Comment by HaraldPanten** [comment]
@JoanMForgeFlow @AaronHForgeFlow could you continue with the migration, please? We need to keep working in other modules depending on this one. 🙏

THX!

**Comment by JoanMForgeFlow** [comment]
> @JoanMForgeFlow @AaronHForgeFlow could you continue with the migration, please? We need to keep working in other modules depending on this one. pray
> 
> THX!

yes!

**Comment by JoanMForgeFlow** [comment]
> Missing changes from #413

@pedrobaeza I am trying to import the odoo_test_helper library, in the test-requirements.txt file, and it gives me the following error. Any idea how to fix it
Captura de pantalla de 2021-12-15 12-34-32
?

**Comment by sbidoul** [comment]
@JoanMForgeFlow just add `odoo-test-helper` in a `test-requirements.txt` file at the repo root.

**Comment by JoanMForgeFlow** [comment]
> @JoanMForgeFlow just add `odoo-test-helper` in a `test-requirements.txt` file at the repo root.

@sbidoul I had already tried it  and it doesn't work, that's why I was asking.

**Comment by sbidoul** [comment]
@JoanMForgeFlow it should work. Please retry and we'll look at the exact log/error.

**Comment by JoanMForgeFlow** [comment]
> @JoanMForgeFlow it should work. Please retry and we'll look at the exact log/error.

@sbidoul It worked!

**Comment by JoanMForgeFlow** [comment]
> Missing changes from #413

@pedrobaeza @AaronHForgeFlow I have already joined the pending PRs, but now I am going to solve the last commit since I have accidentally joined it with the pre-commit.

**Comment by JoanMForgeFlow** [comment]
@sbidoul When github installs the dependencies, it installs the library: odoo-test-helper-1.1.0, but I think it is out of date, how could I specify in the test-requirements.txt file, the latest version of the OCA repository: https://github.com/OCA/odoo-test-helper.

Another question, from where does CI downloads the dependencies written in the test-requirements.txt file.

**Comment by sbidoul** [comment]
> @sbidoul When github installs the dependencies, it installs the library: odoo-test-helper-1.1.0, but I think it is out of date, how could I specify in the test-requirements.txt file, the latest version of the OCA repository: https://github.com/OCA/odoo-test-helper.

You can use the usual git reference syntax: `odoo-test-helper @ git+https://github.com/OCA/odoo-test-helper@master`
I think @simahawk was going to setup an release process for odoo-test-helper.

> Another question, from where does CI downloads the dependencies 

The short answer is: from PyPI.org. A longest answer is it comes from an OCA-hosted index that redirects to PyPI but filters out non-oca Odoo addons, to make sure we test only with OCA dependencies. The [default index is defined in the oca/oca-ci image](https://github.com/OCA/oca-ci/blob/c24a51d3f09c81bfdfa737de9ed16f70ef0a0d7d/Dockerfile#L146-L148).

> written in the test-requirements.txt file.

This applies to all python dependencies (found in manifest external_dependencies, for instance), not only those defined in test-requirements.txt

**Comment by simahawk** [comment]
>You can use the usual git reference syntax: odoo-test-helper @ git+https://github.com/OCA/odoo-test-helper@master
I think @simahawk was going to setup an release process for odoo-test-helper.

Yeah, I still have to get back to it :sweat: 

Meanwhile, IMO it's better if you use the tarball instead:

`odoo-test-helper @ https://github.com/OCA/odoo-test-helper/archive/refs/heads/master.tar.gz`

**Comment by JoanMForgeFlow** [comment]
> > @sbidoul When github installs the dependencies, it installs the library: odoo-test-helper-1.1.0, but I think it is out of date, how could I specify in the test-requirements.txt file, the latest version of the OCA repository: https://github.com/OCA/odoo-test-helper.
> 
> You can use the usual git reference syntax: `odoo-test-helper @ git+https://github.com/OCA/odoo-test-helper@master` I think @simahawk was going to setup an release process for odoo-test-helper.
> 
> > Another question, from where does CI downloads the dependencies
> 
> The short answer is: from PyPI.org. A longest answer is it comes from an OCA-hosted index that redirects to PyPI but filters out non-oca Odoo addons, to make sure we test only with OCA dependencies. The [default index is defined in the oca/oca-ci image](https://github.com/OCA/oca-ci/blob/c24a51d3f09c81bfdfa737de9ed16f70ef0a0d7d/Dockerfile#L146-L148).
> 
> > written in the test-requirements.txt file.
> 
> This applies to all python dependencies (found in manifest external_dependencies, for instance), not only those defined in test-requirements.txt

@sbidoul @simahawk thanks for the quick answer and explanation!. :+1:

**Comment by JoanMForgeFlow** [comment]
@pedrobaeza @sbidoul @simahawk @AaronHForgeFlow @HaraldPanten 

When I run the test I get the following error log. In version 15 model Registry doesn't have 'model_cache' field, and odoo_test_helper is outdated because uses this model. How could we solve this? Thanks in advance

 ```
2021-12-17 08:06:22,023 341 ERROR odoo unittest.suite: ERROR: tearDownClass (odoo.addons.date_range.tests.test_date_range_search_mixin.TestDateRangeearchMixin)
Traceback (most recent call last):
  File "/__w/server-ux/server-ux/setup/date_range/odoo/addons/date_range/tests/test_date_range_search_mixin.py", line 40, in tearDownClass
    cls.loader.restore_registry()
  File "/opt/odoo-venv/lib/python3.8/site-packages/odoo_test_helper/fake_model_loader.py", line 163, in restore_registry
    self.env.registry.model_cache.clear()
AttributeError: 'Registry' object has no attribute 'model_cache'
```

**Comment by HaraldPanten** [comment]
Hi Joan, this question is out of my knowledge. Probably Stéphane or Simone could advise you

**Comment by pedrobaeza** [comment]
Probably `odoo-test-helper` should be updated for supporting v15 cc @simahawk

**Comment by simahawk** [comment]
Here it comes https://github.com/OCA/odoo-test-helper/pull/18

[2.0.0](https://pypi.org/project/odoo-test-helper/2.0.0/) is already available. 2.0.1 will come as the build gets green :wink:

**Comment by simahawk** [comment]
voilá https://pypi.org/project/odoo-test-helper/2.0.1/

**Comment by JoanMForgeFlow** [comment]
> voilá https://pypi.org/project/odoo-test-helper/2.0.1/

Thank you so much :)

**Comment by HaraldPanten** [comment]
@pedrobaeza GO? 👍

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**